### DISCUSSION TITLE
Supported devices

### DISCUSSION THREAD

**Discussion by irarkh** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#300] [General]
Hello, I have a question on supported devices list. Viomi V5 pro, viomi.vacuum.v38, is not in the list. Does this component need mods for working with V5 pro? Or better to say is there any way? I already installed but get no maps. following the issues on same map generation problem I failed to make the component work. Have no idea if its the supported version or there is other issue. Thank you in advanced for answering.

Here is the config and error I got for reference:
```yaml
camera:
  - platform: xiaomi_cloud_map_extractor
    host: 192.168.1.32
    token: 7578504f374fxxxxxxxx
    username: xxxx
    password: xxxx
    draw: ['all']
    attributes:
      - calibration_points
    force_api : viomi
    store_map_raw: true
    store_map_path: "/config"
```
```
Logger: homeassistant.helpers.entity
Source: custom_components/xiaomi_cloud_map_extractor/viomi/vacuum.py:22
Integration: xiaomi_cloud_map_extractor ([documentation](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor), [issues](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/issues))
First occurred: 9:07:12 PM (195 occurrences)
Last logged: 9:24:33 PM

Update for camera.xiaomi_cloud_map_extractor fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 515, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 743, in async_device_update
    raise exc
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/xiaomi_cloud_map_extractor/camera.py", line 278, in update
    self._handle_map_data(map_name)
  File "/config/custom_components/xiaomi_cloud_map_extractor/camera.py", line 335, in _handle_map_data
    map_data, map_stored = self._device.get_map(map_name, self._colors, self._drawables, self._texts,
  File "/config/custom_components/xiaomi_cloud_map_extractor/common/vacuum.py", line 36, in get_map
    map_data = self.decode_map(response, colors, drawables, texts, sizes, image_config)
  File "/config/custom_components/xiaomi_cloud_map_extractor/viomi/vacuum.py", line 22, in decode_map
    unzipped = zlib.decompress(raw_map)
zlib.error: Error -3 while decompressing data: incorrect header check
```
Xiaomi Cloud Map Extractor v2.2.0
Home Assistant Core 2022.6.1
Xiaomi MIoT v1.4.3

**Comment by PiotrMachowski** [comment]
It's possible that your vacuum uses different API, you can try changing following line: `force_api: dreame` or `force_api: roidmi`

**Reply by irarkh** [reply]
Thank you for replaying. I managed to get the map by changing api to roidmi.

**Reply by PiotrMachowski** [reply]
Thanks for info, can you please create an issue to change API for this specific model?

Edit: you don't have to create it anymore

**Reply by Latyn4ik** [reply]
@PiotrMachowski Hi.
Does this component support the Dreame D9 ?

**Reply by PiotrMachowski** [reply]
@Latyn4ik it is on the [list of supported devices](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#supported-devices), so I think it does

**Reply by irarkh** [reply]
> Thanks for info, can you please create an issue to change API for this specific model?
> 
> Edit: you don't have to create it anymore

Sorry I saw your request late.

**Comment by joaopedros2** [comment]
Hello! is it possible to add the ijai.vacuum.v3 model?

**Reply by PiotrMachowski** [reply]
https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/issues/200
